### PR TITLE
[breaking] introduce interface values to rotator yaml unmarshaling

### PR DESCRIPTION
**Pull request by kuannie1**
Current progress:
- realized files weren't marshaling correctly (both sinks and sources)
- basic interfaces implemented but missing tests. I'll need to define custom sinks & sources in config_test.go

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -10,9 +10,30 @@ import (
...
 )
 
+const (
+	flagVerbose = "verbose"
+)
+
 var rootCmd = &cobra.Command{
 	Use:          "rotator",
 	SilenceUsage: true,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
```
Comment (ryanking) [inline cmd/root.go:20]: Might want to switch to the root cmd in go-misc -> https://github.com/chanzuckerberg/go-misc/blob/main/cmds/root.go#L17

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -70,3 +68,68 @@ func TestSingleStringPairs(t *testing.T) {
...
+	return dummyIface, nil
+}
+
+// Kind returns the kind of this source
+func (listSrc *ListSource) Kind() source.Kind {
+	return source.Kind("listSource")
+}
+
+// Todo: Think about whether the dummy source suffices
+func TestConfigWithLists(t *testing.T) {
```
Comment (kuannie1) [inline pkg/config/config_test.go:90]: This test currently fails this way:
```
--- FAIL: TestConfigWithLists (0.00s)
    /Users/anneku/Code/7/rotator/pkg/config/config_test.go:105: 
        	Error Trace:	config_test.go:105
        	Error:      	Expected nil, but got: Unrecognized source
        	Test:       	TestConfigWithLists
FAIL
FAIL	github.com/chanzuckerberg/rotator/pkg/config	1.589s
FAIL
```
Seems like I'd have to add `ListSource` support in secret.MarshalYAML() to keep going

File: pkg/config/config_test.go [hunk 3]
```diff
@@ -70,3 +68,68 @@ func TestSingleStringPairs(t *testing.T) {
...
+
+	c1 := &config.Config{Secrets: []config.Secret{
+		{
+			Name:   "listTest",
+			Source: &ListSource{},
+		},
+	}}
+	// Marshal (just single key-pair values)
+	bytes, err := yaml.Marshal(c1)
+	r.Nil(err)
```
Comment (kuannie1) [inline pkg/config/config_test.go:105]: This test currently fails this way:
```
--- FAIL: TestConfigWithLists (0.00s)
    /Users/anneku/Code/7/rotator/pkg/config/config_test.go:105: 
        	Error Trace:	config_test.go:105
        	Error:      	Expected nil, but got: Unrecognized source
        	Test:       	TestConfigWithLists
FAIL
FAIL	github.com/chanzuckerberg/rotator/pkg/config	1.589s
FAIL
```
Seems like I'd have to add `ListSource` support in secret.MarshalYAML() to keep going

**Comment by kuannie1** [comment]
Closing because this was harder than I thought. The biggest challenge is making the `interface{}` implementation work with [this block](https://github.com/chanzuckerberg/rotator/blob/main/cmd/rotate.go#L108-L126).

**Pull request closed without merging**